ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactScrollView.java

```diff
@@ -1011,6 +1034,80 @@ private void setPendingContentOffsets(int x, int y) {
     }
   }
 
+  private UIManagerModule getUIManagerModule() {
+    ReactContext reactContext = UIManagerHelper.getReactContext(this);
+    return Assertions.assertNotNull(reactContext.getNativeModule(UIManagerModule.class));
+  }
+
+  private void computeTargetViewForMaintainVisibleContentPosition() {
+    if (mMaintainVisibleContentPositionData == null) {
+      return;
+    }
+
+    ReactViewGroup contentView = (ReactViewGroup) mContentView;
+    if (contentView == null) {
+      return;
+    }
+
+    int currentScrollY = getScrollY();
+    int minIndex = mMaintainVisibleContentPositionData.minIndexForVisible;
+    for (int i = minIndex; i < contentView.getChildCount(); i++) {
+      View child = contentView.getChildAt(i);
+      if (child.getY() > currentScrollY || i == contentView.getChildCount() - 1) {
+        mFirstVisibleViewForMaintainVisibleContentPosition = new WeakReference<>(child);
```

Comment:
Do you know why a WeakReference is needed here? I included it in my draft basically copying from an existing draft implementation in an open RN PR, but @Julesssss and I weren't sure if it was actually needed.

Reply:
Not sure, I also kept it because it was there in your implementation. I think this is good to avoid retaining the first visible view in the case it gets removed from the scrollview.